src/Integration.Vsix/InfoBar/InfoBarManager.cs

```diff
@@ -132,6 +172,29 @@ private IInfoBar AttachInfoBarImpl(Guid toolWindowGuid, string message, SonarLin
             return null;
         }
 
+        private IInfoBar AttachInfoBarMainWindowImpl(string message, SonarLintImageMoniker imageMoniker, ButtonStyle buttonStyle = ButtonStyle.Button, params string[] buttonTexts)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            var shell = serviceProvider.GetService(typeof(SVsShell)) as IVsShell;
+            Debug.Assert(shell != null);
+
+            shell.GetProperty((int)__VSSPROPID7.VSSPROPID_MainWindowInfoBarHost, out object value);
+
+            InfoBarModel model = CreateModel(message, buttonTexts, buttonStyle, imageMoniker);
+            IVsInfoBarUIFactory infoBarUIFactory = serviceProvider.GetService(typeof(SVsInfoBarUIFactory)) as IVsInfoBarUIFactory;
+            IVsInfoBarUIElement uiElement;
+
+            if (TryCreateInfoBarUI(infoBarUIFactory, model, out uiElement)
+                && value is IVsInfoBarHost host)
```

Comment:
I'd put the `value is IVsInfoBarHost host` condition first. That way the InfoBarUI won't be created if the condition evaluates to `false`.

It might also be simpler to do the check separately after line 182, and early-out if the returned value isn't a host.